### PR TITLE
test: adding hooks on confirmations e2e wdio and fix failing test

### PR DESCRIPTION
**Pull request by seaona**
## Description
In this PR we introduce the use of wdio hooks for running some background actions/servers (ganache and mocks) before and after Scenarios and removing them from the steps feature files.

This will help as reduce the steps on feature files as well as leverage the conditional tagging for enable/disable background servers/actions in an easier way.

**Note:** Approve ERC20 test is failing due to a change in the Approve flow. We (confirmations) still miss another change in the UI requested by product, so test will be updated after the final flow is working on the app.

## Considerations
1. `beforeScenario` runs before each scenario, and holds during that specific scenario. This has implications in the organization of feature files, as we need to write all the needed steps for that test under the same scenario, otherwise background processes will be stopped after the scenario is over.

2. Cucumber `world` is accessible and set for each scenario and reset after the scenario terminates. This has the same implications as the above point. We need to write all the test steps under the same scenario if we want to access world variables.
On the `beforeScenario` hook from wdio, we can access cucumber world using `context`.

3. We cannot use hooks with `Scenario Outlines` meaning we should always use `Scenario` for tests that use hooks.

## Screenshots

More info about cucumber world state [here](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/world.md)

image

More info about wdio including world (context) in their hooks, which is then passed to cucumber

https://github.com/webdriverio/webdriverio/issues/7651

**Issue**

Progresses #6574  

**Checklist**

* [X] There is a related GitHub issue
* [X] Tests are included if applicable
* [X] Any added code is fully documented

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -296,9 +301,26 @@ export const config = {
...
+  beforeScenario: async function (world, context) {
+    const tags = world.pickle.tags;
+
+    if(tags.filter(e => e.name === '@ganache').length > 0){
+      await startGanache();
+    }
+
+    if(tags.filter(e => e.name === '@multisig').length > 0){
+      const multisig = await deployMultisig();
+      context.multisig = multisig;
```
Comment (seaona) [inline wdio.conf.js:313]: we pass multisig address we just deployed to the cucumber `world` so we can access it later in the scenario steps

File: wdio.conf.js [hunk 2]
```diff
@@ -296,9 +301,26 @@ export const config = {
    * @param {ITestCaseHookParameter} world    world object containing information on pickle and test step
    * @param {Object}                 context  Cucumber World object
    */
-  beforeScenario: ({tags: '@gasApiDown'}, async function (world, context) {
-    context.mock = gasApiDown();
-  }),
+  beforeScenario: async function (world, context) {
```
Comment (seaona) [inline wdio.conf.js:304]: we use conditional hooks depending on the tags for each scenario

File: wdio.conf.js [hunk 3]
```diff
@@ -331,9 +353,17 @@ export const config = {
    * @param {number}                 result.duration  duration of scenario in milliseconds
    * @param {Object}                 context          Cucumber World object
    */
-  afterScenario: ({tags: '@mock'}, async function (world, result, context) {
-    cleanAllMocks();
-  }),
+  afterScenario: async function (world, context) {
```
Comment (seaona) [inline wdio.conf.js:356]: tear down

File: wdio/step-definitions/send-flow.steps.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ Then(/^I tap the Save button/, async () => {
 
 Given(
   /^I enter address "([^"]*)?" in the sender's input box/,
-  async (address) => {
+  async function (address) {
```
Comment (seaona) [inline wdio/step-definitions/send-flow.steps.js:30]: we need to bind `this` for accessing cucumber `world` , so we cannot use arrow functions

File: wdio/features/Confirmations/ApproveERC20.feature [hunk 5]
```diff
@@ -2,25 +2,21 @@
 @confirmations
 @regression
 
-Feature: Approving an ERC20 token from a dapp
+Feature: Approve an ERC20 token from a dapp
   A user should be able to approve an ERC20 token from a dapp.
-
-  Scenario: Import wallet
+  @ganache
+  @erc20
```
Comment (seaona) [inline wdio/features/Confirmations/ApproveERC20.feature:8]: we use tags which will trigger the desired hooks for that particular scenario

File: wdio/features/Confirmations/ApproveERC20.feature [hunk 6]
```diff
@@ -2,25 +2,21 @@
...
     Given the app displayed the splash animation
     And I have imported my wallet
     And I tap No Thanks on the Enable security check screen
     And I tap No thanks on the onboarding welcome tutorial
-  
-  Scenario: Approve an ERC20 token from a dapp with default amount
-    Given Ganache server is started
     And I close the Whats New modal
     And Ganache network is selected
-    And ERC20 token contract is deployed
```
Comment (seaona) [inline wdio/features/Confirmations/ApproveERC20.feature:18]: we can remove background steps as they are now executed with the hooks

File: wdio/features/Confirmations/ApproveERC20.feature [hunk 7]
```diff
@@ -2,25 +2,21 @@
...
     When I navigate to the browser
     And I am on Home MetaMask website
     And I navigate to "test-dapp-erc20"
     And I connect my active wallet to the test dapp
     And I scroll to the ERC20 section
     And I approve ERC20 tokens
-    Then the transaction is submitted with Transaction Complete! toast appearing
-    Then Ganache server is stopped
```
Comment (seaona) [inline wdio/features/Confirmations/ApproveERC20.feature:26]: tear down also happens in the hooks level now

**Review by SamuelSalas — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Pull request merged**